packages/react/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,276 @@
+  ComponentEventHandler,
+  ShorthandValue,
+  WithAsProp,
+  withSafeTypeForAs,
+  Omit,
+} from '../../types'
+import { Accessibility } from '../../lib/accessibility/types'
+import { sliderBehavior } from '../../lib/accessibility'
+import { SupportedIntrinsicInputProps } from '../../lib/htmlPropsUtils'
+import Box from '../Box/Box'
+
+const defaultPropValues = {
+  max: 100,
+  min: 0,
+  step: 1,
+  value: 50,
+}
+
+const processInputValues = (
+  p: Pick<SliderProps, 'min' | 'max'> & Pick<SliderState, 'value'>,
+): { min: number; max: number; value: number; valueAsPercentage: string } => {
+  let min = _.toNumber(p.min)
+  let max = _.toNumber(p.max)
+  let value = _.toNumber(p.value)
+
+  if (isNaN(min)) min = defaultPropValues.min
+  if (isNaN(max)) max = defaultPropValues.max
+  value = isNaN(value) ? defaultPropValues.value : Math.min(max, Math.max(min, value))
+  const valueAsPercentage = `${(100 * (value - min)) / (max - min)}%`
+
+  return { min, max, value, valueAsPercentage }
+}
+
+export interface SliderSlotClassNames {
+  input: string
+  inputWrapper: string
+  rail: string
+  thumb: string
+  track: string
+}
+
+export interface SliderProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    Omit<SupportedIntrinsicInputProps, 'defaultValue'> {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * @default sliderBehavior
+   */
+  accessibility?: Accessibility
+
+  /** The default value of the slider. */
+  defaultValue?: React.ReactText
+
+  /**
+   * A slider can be read-only and unable to change states.
+   * @default false
+   */
+  disabled?: SupportedIntrinsicInputProps['disabled']
+
+  /**
+   * A slider can take the width of its container.
+   * @default false
+   */
+  fluid?: boolean
+
+  /**
+   * Callback that creates custom accessibility message a screen reader narrates when the value changes.
+   * @param {SliderProps} props - Slider props.
+   */
+  getA11yValueMessageOnChange?: (props: SliderProps) => string
+
+  /** Shorthand for the input component. */
+  input?: ShorthandValue
+
+  /** Ref for input DOM node. */
+  inputRef?: React.Ref<HTMLElement>
+
+  /**
+   * The maximum value of the slider.
+   * @default 100
+   */
+  max?: SupportedIntrinsicInputProps['max']
+
+  /**
+   * The minimum value of the slider
+   * @default 0
+   */
+  min?: SupportedIntrinsicInputProps['min']
+
+  /**
+   * Called after item checked state is changed.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onChange?: ComponentEventHandler<SliderProps>
+
+  /**
+   * A number that specifies the granularity that the value must adhere to, or the special value 'any'.
+   * A string value of any means that no stepping is implied, and any value is allowed
+   * (barring other constraints, such as min and max).
+   * @default 1
+   */
+  step?: SupportedIntrinsicInputProps['step']
+
+  /** The value of the slider. */
+  value?: React.ReactText
```

Comment:
This type looks strange... Can we use `number | string` instead?

Reply:
done, thanks